android/src/main/java/com/heapanalytics/reactnative/HeapInitializationProvider.java

```diff
@@ -0,0 +1,50 @@
+package com.heapanalytics.reactnative;
+
+import android.content.ContentProvider;
+import android.content.ContentValues;
+import android.database.Cursor;
+import android.net.Uri;
+
+public class HeapInitializationProvider extends ContentProvider {
+    @Override
+    public boolean onCreate() {
+        String appId;
+
+        if (BuildConfig.DEBUG) {
+            appId = BuildConfig.HEAP_APP_ID_DEV;
+        } else {
+            appId = BuildConfig.HEAP_APP_ID_PROD;
+        }
+
+        if (!appId.isEmpty()) {
+            RNHeap.init(getContext(), appId);
+        }
+
+        return true;
+    }
+
+    @Override
+    public Cursor query(Uri uri, String[] strings, String s, String[] strings1, String s1) {
```

Comment:
Wow, this is an unfortunate API... I realize it's not yours, so not like you can change it, but ick... is this just like pre-collections java or did string arrays come back into favor since I left?

Reply:
Not too sure off the top of my head; the only reason I can think of is performance: maybe they're trying to avoid the extra overhead of using function calls to interact with the collection? I don't really see any other reason to prefer `String[]` over `List<String>`.